.github/workflows/link-check-deploy.yml

```diff
@@ -1,46 +1,57 @@
 name: Check new links against deployment
-# This workflow "triggers" and skips on deployment because GitHub Actions /
-# Checks refuses to show the check on deployment_status
 on:
-  - deployment
   - deployment_status
 jobs:
   run:
-    name: Initialize
+    name: Run
     runs-on: ubuntu-latest
+    continue-on-error: true
     if:
       github.event.deployment.ref != 'master' &&
       github.event.deployment_status.state == 'success'
     steps:
-      - uses: actions/checkout@v2
-      - id: build_check
-        uses: LouisBrunner/checks-action@v1.0.0
+      - uses: actions/checkout@v3
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          name: Report
-          status: queued
+          fetch-depth: 0
       - name: Run Link Check
         id: check
-        uses: iterative/link-check.action@v0.8
+        run: |
+          set +e
+          body="<h1 id=\"link-check\">Link Check Report</h1>
+
+          $(
+            npx repo-link-check \
+            -d master -c config/link-check/config.yml \
+            -r ${{ github.event.deployment.payload.web_url }}
+          )"
+          body="${body//'%'/'%25'}"
+          body="${body//$'\n'/'%0A'}"
+          body="${body//$'\r'/'%0D'}"
+          echo "::set-output name=report::$body"
+          exit 0
+
+      - name: Find PR associated with deployment
+        uses: actions/github-script@v6
+        id: pr-number
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          diff: true
-          configFile: config/link-check/config.yml
-          rootURL: '${{ github.event.deployment.payload.web_url }}'
-          output: checksAction
-      - uses: LouisBrunner/checks-action@v1.0.0
-        if: ${{ success() }}
+          script: |
+            const findPR = require('./.github/workflows/deployment-report.js')
+            return await findPR({ github, context, core })
+
+      - name: Find Existing Link Check Report Comment
+        uses: peter-evans/find-comment@v2
+        id: fc
         with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          check_id: ${{ steps.build_check.outputs.check_id }}
-          status: completed
-          conclusion: ${{ steps.check.outputs.conclusion }}
-          output: ${{ steps.check.outputs.output }}
-      - uses: LouisBrunner/checks-action@v1.0.0
-        if: ${{ failure() }}
+          issue-number: ${{ steps.pr-number.outputs.result }}
+          comment-author: 'github-actions[bot]'
+          body-includes: "<h1 id=\"link-check\">Link Check Report</h1>"
+
+      - name: Create or update comment
+        uses: peter-evans/create-or-update-comment@v2
```

Comment:
@iterative/cml if this action can update a comment without a user-provided PAT, [why can't cml](https://cml.dev/doc/ref/send-comment#github)?

Reply:
@casperdcl  Different kinds of comments, comment on a commit vs comment on a pr?

Reply:
> Different kinds of comments

if that's the issue (a commit-comment requires less perms than pr-comment) then shouldn't `--pr` should fallback to commit-comment + warning rather than just failing?

Reply:
There's [an action that creates commit comments](https://github.com/peter-evans/commit-comment/blob/8e589b3028a09fc36266b90ccf06b679ff74eab9/src/index.js#L32) which, if the docs are to be believed can also take the GHA token

Reply:
opened https://github.com/iterative/cml/issues/1105